### ISSUE TITLE
Crash on startup

### ISSUE THREAD

**Issue by JoshQuest1** [andrew1421lee/ROFL-Player#17]
"Could not find any executables saved, ROFL Player will try to automatically locate League of Legends."

"Exception - System.ID.DirectoryNotFoundException: Input path does not exist
  at
ROFLPlayer.Utilities.GameLOcator.FindLeagueExecutable(String
startingpath)
at
ROFLPlayer.Utilities.ExecsManager.FindAndAddLeagueExec(String startPath)"  

And then closes out.

This does not happen on beta 5 (.8), but started happening on beta 6 (.9)

My league exe is here for reference:
C:\Riot Games\League of Legends\RADS\solutions\lol_game_client_sln\releases\0.0.2.7\deploy

**Comment by fraxiinus** [comment]
Okay, the issue could be because ROFL Player is reading an outdated file location in the registry. You can work around this until I fix it.

1. Create a new file in the execs folder where ROFL Player is.
2. Put the following text into the file, make sure it is all in one line. Save it as "Default.json".
3. Start up ROFL Player and it should detect it as a valid executable and let you in.

`{"Name":"Default","TargetPath":"C:\\Riot Games\\League of Legends\\RADS\\solutions\\lol_game_client_sln\\releases\\0.0.2.7\\deploy\\League of Legends.exe","StartFolder":"C:\\Riot Games\\League of Legends","PatchVersion":"9.6.268.2663","AllowUpdates":true,"IsDefault":true,"ModifiedDate":"2019-03-27T17:55:44.2212944-04:00"}`

**Comment by JoshQuest1** [comment]
Just writing to confirm that the workaround worked.
Thanks!

**Comment by fraxiinus** [comment]
Thanks for confirmation, I'm going to leave this issue open until it is fixed.

**Comment by TheNakriin** [comment]
Hey there,
I've got the same problem and tried the workaround (had to change the path to my other hard drive and used the Version number that you can see in the client, when hovering over "v9.7") but it still gives out the same error.

**Comment by fraxiinus** [comment]
Did you change the target path and the start folder? 
The error occurs when the start folder given does not exist.

**Comment by TheNakriin** [comment]
This is what i wrote in the Default.json file: 

{"Name":"Default","TargetPath":"E:\\Programme\\Riot Games\\League of Legends\\RADS\\solutions\\lol_game_client_sln\\releases\\0.0.2.7\\deploy\\League of Legends.exe","StartFolder":"E:\\Programme\\Riot Games\\League of Legends","PatchVersion":"9.7.18446127.18445799","AllowUpdates":true,"IsDefault":true,"ModifiedDate":"2019-04-14T17:55:44.2212944-04:00"}

So basically, what I did was to change C:\\ to E:\\Programme\\ where i usually have my game folders, such as league is.

**Comment by fraxiinus** [comment]
>  The error occurs when the start folder given does not exist.

Did you actively check that the directory exists?
You also need double slashes in the json file, like `C:\\Programme`

**Comment by TheNakriin** [comment]
For some reason, it doesnt show as two slashes. And yes, i checked it.

**Comment by fraxiinus** [comment]
Can you get me a screenshot of the error exactly, including any dialogs that pop up. Since the directory exists, this error should not be happening.

**Comment by TheNakriin** [comment]
![Here you go](https://user-images.githubusercontent.com/49611863/56101123-1efd7f80-5f21-11e9-85c3-01431b3431b1.png)

**Comment by TheNakriin** [comment]
Actually, since League just told me that they reconfigured some stuff, i rechecked everything. Apparently there is no "0.0.2.7" folder anymore in the releases folder. And there is no "League of Legends.exe" to be found in the "solutions"-folder.

**Comment by fraxiinus** [comment]
Okay, I thought so. Riot is releasing a new patcher and with it comes a new folder structure. The game exe is now located in `\\Riot Games\\League of Legends\\Game\\League of Legends.exe`

See if that works

**Comment by TheNakriin** [comment]
Ok, this is how it looks like now:
{"Name":"Default","TargetPath":"E:\\Programme\\Riot Games\\LeagueOfLegends\\Game\\League of Legends.exe","StartFolder":"E:\\Programme\\Riot Games\\League of Legends","PatchVersion":"9.7.18446127.18445799","AllowUpdates":true,"IsDefault":true,"ModifiedDate":"2019-04-14T17:55:44.2212944-04:00"}

Still doesnt work though, Checked if everything was in place, stuff is where it should be. Maybe it's the Version number that's the problem?

**Comment by fraxiinus** [comment]
Double check your paths. You don't have spaces in the target path, but spaces exist in start folder. The patch version number field is only displayed. It doesn't matter.

**Comment by TheNakriin** [comment]
I fixed it. And it was way easier than I thought, I didnt have the file in the execs folder...

**Comment by fraxiinus** [comment]
Fixed in https://github.com/leeanchu/ROFL-Player/commit/0eb879b3b1a834f66871f1a173b539ad6a58c6c7 and https://github.com/leeanchu/ROFL-Player/commit/9a5dccbf216a02b7beb2df083d55efddf21a5eea

Also changed it so that you can cancel the dialog box and it will still bring you to the settings form so you can manually add an executable. https://github.com/leeanchu/ROFL-Player/commit/e4b510f37fdf956b149a00405842e120c026c08f

Issue will be closed when next version releases with these changes

**Comment by ghost** [comment]
Hey there !
Thank you so much for the ROFL player ! As someone who often review replays, I have been looking for such a tool for so long without success. Your ROFL player is a saving grace.

I got the new patcher today so the ROFL player stopped working. So I found this page and did exactly as you asked. Now the ROFL player works again without any problem and I can check replays and play them.

But a new problem appeared. Whenever I play a replay with ROFL player, the client gives me an error and start a repair the next time I open it (you can see the screenshot). I tried playing replays with the client closed and opened but it didn't change anything. Also, League Director don't detect the replay when opened this way. So maybe it's not the right path?
I'm not sure if this is something you can fix or if it's related to Riot, but I wanted to let you know.

![Screenshot_1](https://user-images.githubusercontent.com/50350165/57270508-e36f5480-708b-11e9-8cf9-27cc0b04563b.png)

Also, do you have a discord for the app or anywhere to contact you except for github? (I made an account just to report this)
I'd like to give you more feedback and suggestions. Your app have so much potential!

Thanks again for taking the time to make such an amazing tool !

**Comment by fraxiinus** [comment]
Hm, interesting problem. ROFLPlayer only makes shortcuts to play replay files, so it's hard to imagine why it's causing the client to repair.

There is no discord server for ROFLPlayer, but I will probably make one in the future. If you prefer to contact me in other ways, you can email me at contact@anchu.dev

I've made a new issue for the bug you are experiencing here: https://github.com/leeanchu/ROFL-Player/issues/19 please use that thread instead of this one. Can you add the exec file you used to that issue?

**Comment by fraxiinus** [comment]
> Fixed in [0eb879b](https://github.com/leeanchu/ROFL-Player/commit/0eb879b3b1a834f66871f1a173b539ad6a58c6c7) and [9a5dccb](https://github.com/leeanchu/ROFL-Player/commit/9a5dccbf216a02b7beb2df083d55efddf21a5eea)
> 
> Also changed it so that you can cancel the dialog box and it will still bring you to the settings form so you can manually add an executable. [e4b510f](https://github.com/leeanchu/ROFL-Player/commit/e4b510f37fdf956b149a00405842e120c026c08f)
> 
> Issue will be closed when next version releases with these changes

Latest release (https://github.com/leeanchu/ROFL-Player/releases/tag/0.9b-beta) now includes these fixes
